ironfish/src/migrations/data/020-wallet-optional-spending-key.ts

```diff
@@ -0,0 +1,120 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { KEY_LENGTH, PUBLIC_ADDRESS_LENGTH } from '@ironfish/rust-nodejs'
+import bufio from 'bufio'
+import { Logger } from '../../logger'
+import { IronfishNode } from '../../node'
+import { IDatabaseEncoding, IDatabaseStore, StringEncoding } from '../../storage'
+import { IDatabase, IDatabaseTransaction } from '../../storage'
+import { Account } from '../../wallet'
+import { Migration } from '../migration'
+
+export class Migration020 extends Migration {
+  path = __filename
+
+  prepare(node: IronfishNode): IDatabase {
+    return node.wallet.walletDb.db
+  }
+
+  async forward(
+    node: IronfishNode,
+    db: IDatabase,
+    tx: IDatabaseTransaction | undefined,
+    logger: Logger,
+  ): Promise<void> {
+    const { walletDb } = node.wallet
+    const accounts: Account[] = []
+
+    const oldAccountStore = this.getOldAccountStore(db)
+    logger.info(`Gathering accounts for migration`)
+    for await (const accountValue of oldAccountStore.getAllValuesIter(tx)) {
+      accounts.push(
+        new Account({
+          ...accountValue,
+          walletDb: node.wallet.walletDb,
+        }),
+      )
+    }
+
+    logger.info(`Making wallet compatible with view only accounts`)
+    await walletDb.db.transaction(async (tx) => {
+      for (const account of accounts) {
+        logger.info('')
+        logger.info(`  Migrating account ${account.name}`)
+        // reserialization of existing wallet will occur in here
+        await walletDb.setAccount(account, tx)
+        logger.info(`  Completed migration for account ${account.name}`)
+      }
+    })
+
+    await oldAccountStore.clear(tx)
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  async backward(): Promise<void> {}
```

Comment:
A reverse migration doesn't make sense here right? If we did a reverse the walletdb would then be incompatible with the current serialization code.

Reply:
usually it's for the purpose that the current code is busted, but yeah we don't need to implement one.